fgspectra/model.py

```diff
@@ -0,0 +1,127 @@
+from abc import ABC, abstractmethod
+import types
+import inspect
+import yaml
+import numpy as np
+
+class Model(ABC):
+    """ Abstract class for model definition
+
+    A model is a class that has one purpose: evaluating the model. What
+    evaluation means is defined by the `eval` method that any child class has to
+    override.
+
+    If the `eval` method of a hypotetical ``Child`` class calls the `eval`
+    method of other `Model`s, it is likely that ``Child`` has also to override
+    `set_defaults`, `defaults` and `_get_repr`
+    """
+
+    def __init__(self, **kwargs):
+        """ You can set defaults at construction time """
+        self.set_defaults(**kwargs)
+
+    def set_defaults(self, **kwargs):
```

Comment:
This method is the tricky part of the PR, it works but is constraining. I could not find a more robust/correct/flexible/elegant solution. I'd be really happy to change, if anybody finds a better one.